app/assets/javascripts/member-facing/redirectors.js

```diff
@@ -0,0 +1,43 @@
+const RegisterMemberRedirector = {
+  attemptRedirect:  function(followUpUrl, member) {
+
+    if(typeof(member) !== 'object') {
+      member = window.champaign.personalization.member;
+    }
+
+    if ( shouldMemberRegister() ) {
+      redirectToRegistration();
+      return true;
+    } else {
+      return false;
+    }
+
+    function shouldMemberRegister() {
```

Comment:
If you write every function as methods, then all methods are public and the intention of how this module should be used is not clear at all. There are a few different ways of obtaining the same behaviour, not sure if there's a preferred one. 
http://www.adequatelygood.com/JavaScript-Module-Pattern-In-Depth.html